src/Microsoft.ML.OnnxTransform/OnnxUtils.cs

```diff
@@ -0,0 +1,404 @@
+            {
+                return new Tensor((System.Int64)(object)data);
+            }
+            else if (typeof(T) == typeof(System.SByte))
+            {
+                return new Tensor((System.SByte)(object)data);
+            }
+            else if (typeof(T) == typeof(System.Int16))
+            {
+                return new Tensor((System.Int16)(object)data);
+            }
+            else if (typeof(T) == typeof(System.UInt32))
+            {
+                return new Tensor((System.UInt32)(object)data);
+            }
+            else if (typeof(T) == typeof(System.UInt64))
+            {
+                return new Tensor((System.UInt64)(object)data);
+            }
+            else if (typeof(T) == typeof(System.UInt16))
+            {
+                return new Tensor((System.UInt16)(object)data);
+            }
+            throw new NotSupportedException($"Unsupported type {typeof(T)}");
+        }
+
+        /// <summary>
+        /// Sonoma API only provides Tensor() constructors with overloaded versions
+        /// based on data type. ML.NET cannot use the overloaded version and requires
+        /// generic version. CreateTensor&lt;T&gt; is generic wrapper on top of
+        /// overloaded Tensor(T[] data, OnnxShape shape) constructors.
+        /// </summary>
+        public static Tensor CreateTensor<T>(T[] data, OnnxShape shape)
+        {
+            if (typeof(T) == typeof(System.Boolean))
+            {
+                return new Tensor(((System.Boolean[])(object)data).ToList(), shape);
+            }
+            else if (typeof(T) == typeof(System.Double))
+            {
+                return new Tensor(((System.Double[])(object)data).ToList(), shape);
+            }
+            else if (typeof(T) == typeof(System.Single))
+            {
+                return new Tensor(((System.Single[])(object)data).ToList(), shape);
+            }
+            else if (typeof(T) == typeof(System.Int32))
+            {
+                return new Tensor(((System.Int32[])(object)data).ToList(), shape);
+            }
+            else if (typeof(T) == typeof(System.Int64))
+            {
+                return new Tensor(((System.Int64[])(object)data).ToList(), shape);
+            }
+            throw new NotImplementedException($"Not implemented type {typeof(T)}");
+        }
+
+        /// <summary>
+        /// Sonoma API only provides CopyTo() functions with overloaded versions
+        /// based on data type. ML.NET cannot use the overloaded version and requires
+        /// generic version. CopyTo&lt;T&gt; is generic wrapper on top of
+        /// overloaded Tensor.CopyTo(List&lt;T&gt; dst) methods.
+        /// Also Tensor.CopyTo(List&lt;T&gt; dst) requires a list input, whereas ML.NET
+        /// provides array buffers to copy values to. This mismatch causes an extra copy.
+        /// </summary>
+        public static void CopyTo<T>(Tensor tensor, T[] dst)
```

Comment:
>public static void CopyTo<T>(Tensor tensor, T[] dst) [](start = 8, length = 52)

you can make it extension method for Tensor class.
```public static void CopyTo<T>(this Tensor tensor, T[] dst)```

so instead of ```OnnxUtils.CopyTo(tensor,values) ```
you can ```tensor.CopyTo